x-pack/filebeat/module/checkpoint/firewall/ingest/pipeline.json

```diff
@@ -0,0 +1,974 @@
+        "field": "checkpoint.mac_destination_address",
+        "target_field": "destination.mac",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.dns_type",
+        "target_field": "dns.question.type",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.domain_name",
+        "target_field": "dns.question.name",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.dns_message_type",
+        "target_field": "dns.type",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.tid",
+        "target_field": "dns.id",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename": {
+        "field": "checkpoint.loguid",
+        "target_field": "event.id",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename": {
+        "field": "checkpoint.sequencenum",
+        "target_field": "event.sequence",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename": {
+        "field": "checkpoint.severity",
+        "target_field": "event.severity",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.action",
+        "target_field": "event.action",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.packet_capture",
+        "target_field": "event.url",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.start_time",
+        "target_field": "event.start",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.first_detection",
+        "target_field": "event.start",
+        "ignore_missing": true,
+        "if": "ctx.event?.start == null"
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.last_detection",
+        "target_field": "event.end",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.app_risk",
+        "target_field": "event.risk_score",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename" : {
+        "field": "checkpoint.file_id",
+        "target_field": "file.inode",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename": {
+        "field": "checkpoint.file_type",
+        "target_field": "file.type",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename": {
+        "field": "checkpoint.file_name",
+        "target_field": "file.name",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename": {
+        "field": "checkpoint.file_size",
+        "target_field": "file.size",
+        "ignore_missing": true
+      }
+    },
+    {
+      "rename": {
+        "field": "checkpoint.file_md5",
+        "target_field": "file.hash.md5",
```

Comment:
if any of the hashes exist they should be added to "related.hash"

Reply:
Done!